code/game/objects/machinery/bots/roomba.dm

```diff
@@ -47,12 +47,35 @@
 ///Called when the roomba moves, sucks in all items held in the tile and sends them to cryo
 /obj/machinery/bot/roomba/proc/suck_items()
 	SIGNAL_HANDLER
+
 	var/sucked_one = FALSE
 	for(var/obj/item/sucker in loc)
 		if(sucker.anchored)
 			continue
 		sucked_one = TRUE
-		sucker.store_in_cryo()
+
+		//Here we try to restock whatever we sucked up
+		var/been_restocked = FALSE
+		for(var/type in GLOB.loadout_linked_vendor[VENDOR_FACTION_NEUTRAL])
+			for(var/datum/vending_product/item_to_restock AS in GLOB.vending_records[type])
+				//Infinite rocket bags are disgusting and you should feel bad
+				if(isstorage(sucker))
+					break //Backpacks go to cryo, shrimple as
```

Comment:
why are we checking this here? We just cycling through these for loops and break repeatedly instead of checking it once. We also already have a flag for whether storage items can be restocked.